var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -263,6 +269,12 @@ def edit(self, spec, prefix):
                 '-ffree-line-length-none',
                 '-ggdb',  # make sure we get proper Fortran backtraces
             ]
+        elif '%aocc' in spec:
+            fcflags += [
+                '-D__F2008',
+                '-ffree-form',
+            ]
+            ldflags += ['-Wl,-z,muldefs']
```

Comment:
Why is this needed? According to documentation this requires that libraries for linking are passed in a specific order. Is this automatically ensured? If yes, how? If not: which libraries are affected?

Reply:
The reason why I am insisting on this is that with libraries like COSMA, we already have an overloading of `p*gemm`, depending on how it is linked and it is hard to figure out which implementation is actually called when a user reports a performance regression. Hence we have to be able to reason about which implementation is called and why.
Also, static linking can make this even more complicated.

Reply:
No performance regression observed without `ldflags += ['-Wl,-z,muldefs']` hence removing the same.